src/librustc/ty/mod.rs

```diff
@@ -2675,12 +2667,23 @@ fn associated_item_def_ids<'a, 'tcx>(tcx: TyCtxt<'a, 'tcx, 'tcx>,
     Rc::new(vec)
 }
 
+fn def_span<'a, 'tcx>(tcx: TyCtxt<'a, 'tcx, 'tcx>, def_id: DefId) -> Span {
+    match tcx.hir.span_if_local(def_id) {
+        Some(span) => span,
+        None => {
+            let node_id = tcx.sess.cstore.item_body(tcx, def_id).id().node_id;
```

Comment:
I'm surprised you found `item_body` - that's really only meant to be used by const evaluation.
You should really be *panicking* if the `DefId` isn't local (i.e. you can `.unwrap()` the result of `span_if_local` instead of handling the `None` case) - because the line you added in `src/librustc_metadata/cstore_impl.rs` is what actually handles that case.